src/Components/ImagesTable/Target.tsx

```diff
@@ -0,0 +1,54 @@
+import {
+  useGetClonesQuery,
+  useGetComposeStatusQuery,
+} from '../../store/apiSlice';
+
+type TargetProps = {
+  composeId: string;
+};
+
+const targetOptions = {
+  aws: 'Amazon Web Services',
+  'aws.s3': 'Amazon Web Services',
+  azure: 'Microsoft Azure',
+  gcp: 'Google Cloud Platform',
+  vsphere: 'VMWare',
+  'guest-image': 'Virtualization - Guest image',
+  'image-installer': 'Bare metal - Installer',
+  'edge-commit': 'edge-commit',
+  'edge-installer': 'edge-installer',
+  ami: 'ami',
+  'rhel-edge-commit': 'rhel-edge-commit',
+  'rhel-edge-installer': 'rhel-edge-installer',
+  vhd: 'vhd',
+};
+
+const Target = ({ composeId }: TargetProps) => {
+  const { data: clones, isSuccess: isSuccessClones } =
+    useGetClonesQuery(composeId);
+  const { data: composeStatus, isSuccess: isSuccessTarget } =
+    useGetComposeStatusQuery(composeId);
+  const target = composeStatus?.request?.image_requests[0];
+
+  let tar;
+
+  if (isSuccessTarget) {
+    if (target?.upload_request?.type === 'aws.s3') {
+      tar = targetOptions[target?.image_type];
+    } else if (target?.upload_request?.type === 'aws') {
+      if (isSuccessClones) {
+        tar =
+          targetOptions[target?.upload_request?.type] +
+          ` (${clones.meta.count !== 0 ? clones.meta.count + 1 : 1})`;
+      }
+    } else {
+      tar = targetOptions[target?.upload_request?.type];
+    }
+  }
+  if (tar) {
+    return tar;
+  }
+  return targetOptions[target?.image_type] || null;
```

Comment:
Note that we never returned null: ` return <>{target ? target : compose.imageType}</>;`. Default to the raw `image_type`. In case we don't have the type in `targetOptions`.

Reply:
I've just pushed a PR (https://github.com/RedHatInsights/image-builder-frontend/pull/1247) with updates to our fixtures. With the changes this PR should pass tests. 

What we should probably still do though is try to avoid returning null as that could cause unforeseen problems later. Either with something like this:
```
if (target) {
	return target
} else {
	return composeStatus?.request?.image_requests[0]?.image_type
}
```
or with a ternary operator similar to the one we've used before:
```
return <>{target ? target : composeStatus?.request?.image_requests[0]?.image_type}</>;
```

The problem is both break the tests as of now even though everything renders without any problems. Still if we need to "bend" something, I feel it should rather be tests than the code. What do you think?

I'll continue looking into the failing tests.